adapter/denier/denier.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2017 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package denier
```

Comment:
> If an import is attempted using a path that is not canonical, the go command will refuse to compile the importing package.

- https://golang.org/doc/go1.4#canonicalimports

This basically will give us the option to move the code in the unlikely event github disappears by ensuring it isn't ever used as "github.com/istio/mixer/adapter/denier" or whatever.